drivers/flash/nrf_qspi_nor.c

```diff
@@ -426,11 +427,22 @@ static int qspi_wait_while_writing(const struct device *dev)
 	return (ret < 0) ? ret : 0;
 }
 
-static int qspi_wrsrX(const struct device *dev, uint8_t sr_val, uint8_t sr_num)
+static int qspi_wrsrX(const struct device *dev, uint16_t sr_val, uint8_t sr_num)
```

Comment:
I was thinking in here, we could do this as a read-modify-write for the QER modes that write sr2 along with sr1 at the same time.

Essentially if writing sr1, read sr2 then write the whole thing. If writing sr2, read sr1 then write the whole thing (to accomodate S2B1v1, where writing sr1 clears the QER bits in sr2).

That would keep it consistent with rdsrX, where it outputs a single byte, this would always take in a single byte and a register number, and handle the read-modify-write as needed internally.

As a nice side-effect there, all of the S2B1 variants just call wrsrX with the sr2 contents, and this function would handle the specifics of "how" to write sr2.

Reply:
Sorry, I am not sure I completely follow what is written. Are you saying basically we move the if logic I put into `qspi_nrfx_configure` here, so that if the variant needs the first byte it appropriately reads in in here, and then writes in the manor of that spec?

If so, I think that is fine, I considered it myself, but it really is just a question on personal taste of function organization, if a `write` function should have side effects, or just `write` leaving the side effects elsewhere, but as you say it does open up to different modes of doing it cleanly from the outside.

I have no preference, if you think that is better I am for it, and I don't really mind if you want to pull this and add a commit, or if you would like me to do it, with the caveat I will be away from my computer for the next while, so if you want to make the change go for it.

Reply:
FYI I did test this as is, and it does work (as expected since it was similar to my other stuff) So as long as the end result does the same, it should work on my board, but I will test when I can.